src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -963,6 +1007,67 @@ private static void updateVelocity(Entity e, double scale)
             }
             if (e instanceof LivingEntity) ((LivingEntity) e).setHealth(health);
         });
+
+        put("may_fly", (e, v) -> {
+            boolean mayFly = v.getBoolean();
+            if (e instanceof ServerPlayerEntity player) {
+                player.getAbilities().allowFlying = mayFly;
+                if (!mayFly && player.getAbilities().flying) {
+                    player.getAbilities().flying = false;
+                }
+                player.sendAbilitiesUpdate();
+            }
+        });
+
+        put("flying", (e, v) -> {
+            boolean flying = v.getBoolean();
+            if (e instanceof ServerPlayerEntity player) {
+                player.getAbilities().flying = flying;
+                player.sendAbilitiesUpdate();
+            }
+        });
+
+        put("may_build", (e, v) -> {
+            boolean mayBuild = v.getBoolean();
+            if (e instanceof ServerPlayerEntity player) {
+                player.getAbilities().allowModifyWorld = mayBuild;
+                player.sendAbilitiesUpdate();
+            }
+        });
+
+        put("insta_build", (e, v) -> {
+            boolean instaBuild = v.getBoolean();
+            if (e instanceof ServerPlayerEntity player) {
+                player.getAbilities().creativeMode = instaBuild;
+                player.sendAbilitiesUpdate();
+            }
+        });
+
+        put("fly_speed", (e, v) -> {
+            Float flySpeed = NumericValue.asNumber(v).getFloat();
+            if (e instanceof ServerPlayerEntity player) {
+                player.getAbilities().setFlySpeed(flySpeed);
+                player.sendAbilitiesUpdate();
```

Comment:
Any reason why you are using the boxed (uppercase F) float here and in the ones below? You should probably use primitive `float`.